### ISSUE TITLE
Incorrect horizons flag in EDDN messages from CAPI updates

### ISSUE THREAD

**Issue by HansAcker** [EDCD/EDMarketConnector#2277]
**Please complete the following information:**

[//]: # (You can gather most of this information with the EDMC System Profiler)
 
 - Version: 5.11.2
 - Game Version: Live Horizons
 - OS: Win 10
 - OS Locale: eng

Commodity market updates from CAPI in Live-Horizons in _some_ cases produce messages with ``horizons: false, odyssey: false``. Often seen at fleet carriers and planetary outposts, not at space stations.

It looks like capi_is_horizons() incorrectly returns false when a station offers no ships or modules and has no Colony economy?
I don't quite follow the logic, tbh.

**To Reproduce**

1. Log into Horizons (Live)
2. Dock to a fleet carrier
3. Often, this already produces a market message with horizons: false
4. Re-log
5. That reliably produces a Location message with horizons: true and a market update with horizons: false
6. Updating market.json produces a message with the correct flags

**Additional context**

Messages after logging into Horizons while docked at a carrier.
Location:
```
{
  "$schemaRef": "https://eddn.edcd.io/schemas/journal/1",
  "header": {
    "gamebuild": "r303387/r0 ",
    "gameversion": "4.0.0.1807",
    "gatewayTimestamp": "2024-07-19T11:08:23.337637Z",
    "softwareName": "E:D Market Connector [Windows]",
    "softwareVersion": "5.11.2",
    "uploaderID": "a4b7fa87b574c8ec74ab6b0ac45ab9a6eebbebb3"
  },
  "message": {
...
    "StationEconomies": [
      {
        "Name": "$economy_Carrier;",
        "Proportion": 1
      }
    ],
...
    "StationName": "H7B-B1M",
    "StationServices": [
      "dock",
      "autodock",
      "commodities",
      "contacts",
      "exploration",
      "crewlounge",
      "rearm",
      "refuel",
      "repair",
      "engineer",
      "flightcontroller",
      "stationoperations",
      "stationMenu",
      "carriermanagement",
      "carrierfuel",
      "socialspace",
      "bartender",
      "vistagenomics"
    ],
    "StationType": "FleetCarrier",
...
    "Taxi": false,
    "event": "Location",
    "horizons": true,
    "odyssey": false,
    "timestamp": "2024-07-19T11:08:22Z"
  }
}
```

Market update from CAPI with both flags set to false:
```
{
  "$schemaRef": "https://eddn.edcd.io/schemas/commodity/3",
  "header": {
    "gamebuild": "",
    "gameversion": "CAPI-Live-market",
    "gatewayTimestamp": "2024-07-19T11:08:29.271407Z",
    "softwareName": "E:D Market Connector [Windows]",
    "softwareVersion": "5.11.2",
    "uploaderID": "a4b7fa87b574c8ec74ab6b0ac45ab9a6eebbebb3"
  },
  "message": {
... 
   "horizons": false,
    "odyssey": false,
    "timestamp": "2024-07-19T11:08:28Z"
  }
}
```

Correct flags on event from local file:
```
{
  "$schemaRef": "https://eddn.edcd.io/schemas/commodity/3",
  "header": {
    "gamebuild": "r303387/r0 ",
    "gameversion": "4.0.0.1807",
    "gatewayTimestamp": "2024-07-19T11:10:12.475493Z",
    "softwareName": "E:D Market Connector [Windows]",
    "softwareVersion": "5.11.2",
    "uploaderID": "a4b7fa87b574c8ec74ab6b0ac45ab9a6eebbebb3"
  },
  "message": {
...
    "horizons": true,
    "odyssey": false,
    "timestamp": "2024-07-19T11:10:11Z"
  }
}
```

Market update at a station with shipyard and outfitting:
```
{
  "$schemaRef": "https://eddn.edcd.io/schemas/commodity/3",
  "header": {
    "gamebuild": "",
    "gameversion": "CAPI-Live-market",
    "gatewayTimestamp": "2024-07-19T11:49:07.196899Z",
    "softwareName": "E:D Market Connector [Windows]",
    "softwareVersion": "5.11.2",
    "uploaderID": "07cb70c2ad8eb4f2b0d920bb71fff83341a0308d"
  },
  "message": {
    "horizons": true,
    "odyssey": false,
    "stationName": "Sharp Dock",
    "systemName": "Andel",
    "timestamp": "2024-07-19T11:49:06Z"
  }
}
```

**Please Confirm the Following...**

[//]: # (Add any other context about the problem here.)
- [x] I have checked the [Known Issues](https://github.com/EDCD/EDMarketConnector/issues/618) list to ensure this is not a duplicate
- [ ] I have checked the [Troubleshooting Guide](https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting) to check for known workarounds

**Comment by klightspeed** [comment]
It looks like [`capi_is_horizons`](https://github.com/EDCD/EDMarketConnector/blob/main/plugins/eddn.py#L2641-L2689) wasn't updated for the universe split or for carriers. I wonder if there are any cases even in the legacy universe where someone can have no access to Horizons?

**Comment by Eahlstan** [comment]
>I wonder if there are any cases even in the legacy universe where someone can have no access to Horizons?

Using `Legacy Elite Dangerous` instead of `Legacy Horizons` can do that. (Useful if you're stuck on a planet.)

**Comment by Athanasius** [comment]
It's up to EDDN Listeners to know the semantics behind the `horizons` and `odyssey` flags.

As outlined in https://github.com/EDCD/EDDN/blob/master/docs/Developers.md#horizons-and-odyssey-flags the only reliable source is Journal events, but, and this is **very important**, *you can't assume that CAPI data retrieved for an in-game player is definitely going to match the state of the Journals for that session*.  So you have to attempt using something in the CAPI data to set these flags.

In the case of `shipyard` and `outfitting` messages a `horizons` flag can be synthesized from the present of any ship or module with a `HORIZONS_SKU` being present.  There's also something about seeing a `Colony` economy type.  I think that's because those are only present on surface ports.  That's got nothing to do with the recently added player colonisation. **Other than that all CAPI-sourced messages cannot have reliable `horizons` or `odyssey` flags**.

Or at least that was the state of things when I last reviewed all of this 2022-09-27.  It's possible something has changed since.

TL;DR - Stop worrying about the horizons and odyssey flags on CAPI-sourced commodities messages, they're not going to tell you anything useful.

**Comment by Rixxan** [comment]
I can throw in another best-effort check for the function, but agree that the flags cannot be reliably determined from CAPI messages.

**Comment by Athanasius** [comment]
EDDN Listeners correct me if I'm wrong, but I believe the only use case for the `horizons` flag is for scenarios like:

1. Someone with Horizons docks and sends the data, a module or ship requiring Horizons is sold.
2. Someone without Horizons docks and send the data, but because they don't have access those ships/modules aren't in it.

The flag allows a Listener to go "no Horizons access stated, so maybe I don't remove them from the 'available' list because we don't know for sure".

In the FC case, unless one of those pre-set ship packages is for a Horizons-only ship, or has Horizons-only modules, the entire thing is moot for the `shipyard` and `outfitting` messages.  It's always moot for `commodities` messages because there's nothing Horizons-only there (in the sense of with/without Cmdrs seeing something different, obviously a without Cmdr could never even get to a market with surface-only commodities).

**Comment by Rixxan** [comment]
Since this doesn't seem to have an impact, and on discussions internally, it sounds like this was intended behavior and as such (outside of guidance from EDDN's team specifically) no fix is warranted. 

Closing as wontfix for now, but that may change in the future if updated guidance is received.
